aviary/examples/external_subsystems/engine_NPSS/NPSS_Model/DesignEngineGroup.py

```diff
@@ -0,0 +1,167 @@
+        Fn_SLS = float(parser.transfer_var(0, 3))
+        outputs['Fn_SLS'] = Fn_SLS
+
+        parser.mark_anchor("Wf_training_data")
+        parser.set_delimiters(', ')
+        Wf_training_data_raw = parser.transfer_array(0, 4, 0, vec_size*3)
+        Wf_training_data_snip = [
+            a for a in Wf_training_data_raw if a.replace('.', '').isnumeric()]
+        Wf_training_data = np.array(Wf_training_data_snip, float)
+        outputs['Wf_training_data'] = Wf_training_data
+
+        parser.mark_anchor("thrust_training_data")
+        parser.set_delimiters(', ')
+        thrust_training_data_raw = parser.transfer_array(0, 4, 0, vec_size*3)
+        thrust_training_data_snip = [
+            a for a in thrust_training_data_raw if a.replace('.', '').isnumeric()]
+        thrust_training_data = np.array(thrust_training_data_snip, float)
+        outputs['thrust_training_data'] = thrust_training_data
+
+        parser.mark_anchor("thrustmax_training_data")
+        parser.set_delimiters(', ')
+        thrustmax_training_data_raw = parser.transfer_array(0, 4, 0, vec_size*3)
+        thrustmax_training_data_snip = [
+            a for a in thrustmax_training_data_raw if a.replace('.', '').isnumeric()]
+        thrustmax_training_data = np.array(thrustmax_training_data_snip, float)
+        outputs['thrustmax_training_data'] = thrustmax_training_data
+        # print(thrustmax_training_data)
+
+
+class DesignEngineGroup(om.Group):
+    def setup(self):
+        self.add_subsystem('DESIGN', NPSSExternalCodeComp(),
+                           promotes_inputs=[('W_DES', Aircraft.Engine.DESIGN_MASS_FLOW)],
+                           # promotes_outputs=[('Fn_SLS', Aircraft.Engine.SCALED_SLS_THRUST), ('thrust_training_data', Dynamic.Mission.THRUST+'_train'),
+                           #                   ('thrustmax_training_data', Dynamic.Mission.THRUST_MAX+'_train'), ('Wf_training_data', 'Wf_td')])
+                           promotes_outputs=[('Fn_SLS', Aircraft.Engine.SCALED_SLS_THRUST), ('thrust_training_data', 'Fn_train'),
+                                             ('thrustmax_training_data', 'Fn_max_train'), ('Wf_training_data', 'Wf_td')])
+
+        self.add_subsystem('neg_Wf', om.ExecComp('y=-x',
+                                                 x={'val': np.ones(
+                                                     vec_size), 'units': 'lbm/s'},
+                                                 y={'val': np.ones(vec_size), 'units': 'lbm/s'}),
+                           promotes_inputs=[('x', 'Wf_td')],
+                           promotes_outputs=[('y', 'Wf_inv_train')])
+
+    def configure(self):
+        self.set_input_defaults(Aircraft.Engine.DESIGN_MASS_FLOW, 240.0, units='lbm/s')
+        self.set_input_defaults('DESIGN.Alt_DES', 0.0, units='ft')
+        self.set_input_defaults('DESIGN.MN_DES', 0.0)
+        super().configure()
+
+
+if __name__ == "__main__":
+    import openmdao.api as om
+
+    SNOPT_EN = False
```

Comment:
We don't generally (anymore) put code down in these blocks, because it is untested and tends to get scale. This might be a good idea to add as a test though. You should add a test regardless, even if it doesn't assert any values, but just makes sure this still runs without error as we continue to develop aviary.

Reply:
I am not sure how to proceed with this comment. I was looking at the other external_subystem examples and there is no test folder.  How are these tested?  For my commit, how does that work when half the code is NPSS? I am guessing the runners that run the tests do not have NPSS on the machines.

Reply:
I just took a look at the examples. It looks like they are all tested here: https://github.com/OpenMDAO/Aviary/blob/main/aviary/examples/test/test_all_examples.py

That means all you have to do is have a script that has a name beginning with "run_".  I see that you do have one, so it is already being tested (though it is failing right now).

Reply:
I removed the test and uploaded.  Not sure if i need to make another test.

Reply:
I added a bench test.

Reply:
bench test should be working now.